src/data_insertion.py

```diff
@@ -109,7 +109,7 @@ def data_embed_insertion(pinecone_index, data, start_from):
                 individual_data_insertion(pinecone_index, [embed], data[i])
             elif isinstance(text, str) and "This model's maximum context length is 8191 tokens" in str(e):
                 embed = []
-                text_chunk = split_text(text, max_words_per_chunk=1000)
+                text_chunk = split_text(text, max_tokens=MAX_TOKENS_PER_CHUNK // 2)
```

Comment:
How are we determining `max_tokens` here exactly to be `MAX_TOKENS_PER_CHUNK // 2` Let's discuss it a bit.

Reply:
Whenever there's an error where maximum token length is exceeded, previously I noticed the chunk_size was reduced by 2.5x. This was required since words was the quanta of length instead of exact tokens. When we move to token-level chunking, this error should in theory never arise (given we specify the correct chunk size for embeddings specified in the model) but I did this just to be safe for now.

Reply:
ok makes sense.